### PR TITLE
Voice

### PR DESCRIPTION
**Pull request by jaanonim**

### CODE REVIEW

**Review by Dudek03 — APPROVED** [review]

File: modules/voice/bot.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    async def cleanup(self, guild):
+        try:
+            await guild.voice_client.disconnect()
+        except AttributeError:
+            pass
+
+        try:
+            del self.players[guild.id]
+        except KeyError:
+            pass
```
Comment (n2oneProgrammer) [inline modules/voice/bot.py]: I don't like that. 
I think if quild.id in  self.players:
del self.players[guild.id] is better

File: modules/voice/bot.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            try:
+                await channel.connect()
+            except asyncio.TimeoutError:
+                raise DiscordException(
+                    f"Connecting to channel: <{channel}> timed out."
+                )
+
+    async def play(self, ctx, search, count):
+        player = self.get_player(ctx)
+        sources = await YTDLSource.create_source(ctx.author, search, loop=self.bot.loop, count=count)
```
Comment (n2oneProgrammer) [inline modules/voice/bot.py]: split it, please?

File: modules/voice/music_player.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        self.bot = ctx.bot
+        self._guild = ctx.guild
+        self._channel = ctx.channel
+        self._cog = ctx.cog
+        self._ui = ui
+
+        self.queue = Queue()
+        self.next = asyncio.Event()
+
+        self.np = None  # Now playing message
```
Comment (n2oneProgrammer) [inline modules/voice/music_player.py]: kendra moment

File: modules/voice/views/add_modal.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    )
+
+    async def on_submit(self, interaction: discord.Interaction):
+        embed = await self.ui.music_cog.play(self.ui.ctx, self.search.value, 1)
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+        await self.ui.update()
+
+    async def on_error(self, interaction: discord.Interaction, error: Exception) -> None:
+        logging.getLogger("discord").error(error, exc_info=True)
+        await self.ui.update()
```
Comment (n2oneProgrammer) [inline modules/voice/views/add_modal.py:25]: it's look useful for other commands
can you move to move global directory

File: utils/module.py [hunk 5]
```diff
@@ -27,9 +29,12 @@ def get_all_modules(src: str) -> list[str]:
...
+    except TypeError as e:
+        print(e)
         return None
-    except ModuleNotFoundError:
+    except ModuleNotFoundError as e:
+        print(e)
         return None
-    except AttributeError:
+    except AttributeError as e:
+        print(e)
```
Comment (n2oneProgrammer) [inline utils/module.py]: And know you have a lots of useless consoling errors

**Review by n2oneProgrammer — CHANGES_REQUESTED** [review]
Sorry

File: modules/voice/music_player.py (hunk 3 above)
Comment (jaanonim) [reply modules/voice/music_player.py]: it's copyed :-)

File: modules/voice/views/add_modal.py (hunk 4 above)
Comment (jaanonim) [reply modules/voice/views/add_modal.py:25]: It isn't. It's specific use for Modal provided by discord.py

Comment (n2oneProgrammer) [reply modules/voice/views/add_modal.py:25]: o nevermind

File: modules/voice/music_player.py (hunk 3 above)
Comment (n2oneProgrammer) [reply modules/voice/music_player.py]: it's not an excuse.

**Review by n2oneProgrammer — APPROVED** [review]
LGTM

**Pull request merged**